docs/pages/transactions/using-eip-7702/third-party.mdx

```diff
@@ -0,0 +1,76 @@
+If you have an existing custom signer or another third-party embedded wallet provider, you can upgrade your embedded EOAs to our smart wallets by connecting your existing signer.
+
+**Requirements**
+Your signer or embedded wallet provider must support `signAuthorization` in order to be compatible with EIP-7702 delegations.
+To bring your own signer, you'll create a `SmartAccountSigner` interface that implements `signAuthorization`.
+See the details for the interface requirements [here](/wallets/third-party/signers#implementing-smartaccountauthenticator-or-smartaccountsigner).
+
+For example, you can upgrade an existing Privy embedded EOA by extending `WalletClientSigner` to include the `signAuthorization` method exposed by Privy.
+
+<CodeBlocks>
+
+```ts twoslash example.ts
+import {
+  AuthorizationRequest,
+  SmartAccountSigner,
+  WalletClientSigner,
+} from "@aa-sdk/core";
+import { createWalletClient, custom } from "viem";
+import { sepolia } from "viem/chains";
+import { SignedAuthorization } from "viem/experimental";
+import { useSignAuthorization, useWallets } from "@privy-io/react-auth";
+
+const { wallets } = useWallets();
+const embeddedWallet = wallets.find((x) => x.walletClientType === "privy"); // or any other EIP-1193 provider
+const { signAuthorization } = useSignAuthorization();
```

Comment:
wait, is this supposed to just a vanilla js example or a react example? looks like react since you are using these hooks, so should this be within a component?

also, if it's react, ideally we don't re-create all of the objects below on every render. but maybe it's no big deal just for an example to keep it simple. but if you want to optimize it a little, you can use state w/ an init function, i.e.
```tsx
const [signer] = useState<WalletClientSigner>(() => new WalletClientSigner(...))
```